### PR TITLE
Adds VVWriteable to several components

### PR DESCRIPTION
**Pull request by LordEclipse**
It just makes sense that these components should have it.

### CODE REVIEW

File: Content.Shared/Nutrition/Components/SmokableComponent.cs [hunk 1]
```diff
@@ -13,16 +13,16 @@ public sealed partial class SmokableComponent : Component
         /// <summary>
         ///     Solution inhale amount per second.
         /// </summary>
-        [DataField("inhaleAmount")]
+        [DataField("inhaleAmount"), ViewVariables(VVAccess.ReadWrite)]
         public FixedPoint2 InhaleAmount { get; private set; } = FixedPoint2.New(0.05f);
 
-        [DataField("state")]
+        [DataField("state"), ViewVariables(VVAccess.ReadWrite)]
```
Comment (lzk228) [inline Content.Shared/Nutrition/Components/SmokableComponent.cs:19]: dont really think it's necessary for sprites

Comment (deltanedas) [reply Content.Shared/Nutrition/Components/SmokableComponent.cs:19]: also it doesnt update appearance so it wouldnt do anything

File: Content.Shared/Weather/WeatherComponent.cs [hunk 2]
```diff
@@ -38,7 +38,7 @@ public sealed partial class WeatherData
     [ViewVariables(VVAccess.ReadWrite), DataField("endTime", customTypeSerializer: typeof(TimeOffsetSerializer))]
     public TimeSpan? EndTime;
 
-    [ViewVariables]
+    [ViewVariables(VVAccess.ReadWrite)]
     public TimeSpan Duration => EndTime == null ? TimeSpan.MaxValue : EndTime.Value - StartTime;
```
Comment (metalgearsloth) [inline Content.Shared/Weather/WeatherComponent.cs:42]: This isn't settable.

Comment (LordEclipse) [reply Content.Shared/Weather/WeatherComponent.cs:42]: Ah okay.

**Pull request merged**